test/demo/createNotification.inttest.ts

```diff
@@ -0,0 +1,167 @@
+            const cardTitle = await titleElement!.getProperty('innerHTML').then(val => val.jsonValue());
+
+            expect(cardTitle).toEqual(note.title);
+        });
+    });
+
+    describe('With the notification center showing', () => {
+        beforeAll(async () => {
+            // Show the center to ensure we don't get toasts
+            if (!(await isCenterShowing())) {
+                await notifsRemote.toggleNotificationCenter({uuid: 'test-app', name: 'test-app'});
+            }
+        });
+
+        let testApp: Application;
+        let testWindowIdentity: Identity;
+        beforeEach(async () => {
+            testApp = await createTestApp();
+            testWindowIdentity = await testApp.getWindow().then(w => w.identity);
+        });
+
+        afterEach(async () => {
+            testApp.close();
+        });
+
+        test('No toast is shown for the created notification', async () => {
+            const note = await notifsRemote.create(testWindowIdentity, validOptions);
+            await delay(1000);
+
+            const toastWindow = await getToastWindow(testApp.identity, note.id);
+            expect(toastWindow).toBe(undefined);
+
+            await notifsRemote.clear(testWindowIdentity, note.id);
+        });
+
+        describe('When passing invalid options', () => {
+            let createPromise: Promise<Notification>;
+            beforeEach(() => {
+                // Intentionally circumventing type check with cast for testing purposes
+                createPromise = notifsRemote.create(testWindowIdentity, {id: 'invalid-notification'} as NotificationOptions);
+            });
+
+            afterEach(async () => {
+                // Cleanup the created notification (just in case it does get made)
+                await createPromise.catch(() => {});
+                await notifsRemote.clear(testWindowIdentity, 'invalid-notification');
+            });
+
+            test('An error is thrown', async () => {
+                expect(createPromise).rejects.toThrow();
+            });
+
+            test('No notification is created', async () => {
+                // We want to be sure the action has completed, but don't care about the result
+                await createPromise.catch(() => {});
+
+                // Should be no card in the center for this ID
+                const noteCards = await getCardsByNotification(testWindowIdentity.uuid, 'invalid-notification');
+                expect(noteCards.length).toBe(0);
+            });
+        });
+
+        describe('With no ID specified', () => {
+            let createPromise: Promise<Notification>;
+            beforeEach(async () => {
+                createPromise = notifsRemote.create(testWindowIdentity, validOptions);
+            });
+
+            afterEach(async () => {
+                try {
+                    const note = await createPromise;
+                    await notifsRemote.clear(testWindowIdentity, note.id);
```

Comment:
How robust does this clean-up need to be? There will be no `clear` call if the initial promise rejects (which it should).

Reply:
If the initial promise rejects there should be no notification to clear, but I guess we can just to separate catches